Include/internal/pycore_object.h

```diff
@@ -280,6 +289,64 @@ extern PyStatus _PyObject_InitState(PyInterpreterState *interp);
 extern void _PyObject_FiniState(PyInterpreterState *interp);
 extern bool _PyRefchain_IsTraced(PyInterpreterState *interp, PyObject *obj);
 
+static inline void
+_Py_INCREF_TYPE(PyTypeObject *type)
+{
+#ifndef Py_GIL_DISABLED
+    Py_INCREF(type);
+#else
+    if (!_PyType_HasFeature(type, Py_TPFLAGS_HEAPTYPE)) {
+        assert(_Py_IsImmortal(type));
+        return;
+    }
+
+    _PyThreadStateImpl *tstate = (_PyThreadStateImpl *)_PyThreadState_GET();
+    PyHeapTypeObject *ht = (PyHeapTypeObject *)type;
+
+    // Unsigned comparison so that `ht_id=-1` is treated as out-of-bounds.
+    Py_ssize_t ht_id = ht->_ht_id;
```

Comment:
Nit comment:
Can we extract those two common functions as
- get_heap_type_id(ht);
- check_heap_type_id_assigend(ht_id);

It will be helpful to understand what's happening in here for people like me :)

Reply:
By the way, you can ignore this, but some people may need to understand the mechanism when they have to write similar features. At that time it will be help to verify ht_id

Reply:
I don't see a great way of refactoring out this check -- I think a function name that accurately describes the check will be awkwardly long. It's not just that the heap type has an assigned id (i.e., `ht_id != -1`), but also that the thread state has space for it its per-thread array (i.e., `ht_id < size`).